state-sync/aptos-data-client/src/aptosnet/mod.rs

```diff
@@ -434,6 +457,35 @@ impl AptosDataClient for AptosNetDataClient {
         Ok(response.map(|epoch_change| epoch_change.ledger_info_with_sigs))
     }
 
+    async fn get_new_transaction_outputs_with_proof(
+        &self,
+        known_version: Version,
+        known_epoch: Epoch,
+    ) -> Result<Response<(TransactionOutputListWithProof, LedgerInfoWithSignatures)>> {
+        let request = StorageServiceRequest::GetNewTransactionOutputsWithProof(
+            NewTransactionOutputsWithProofRequest {
+                known_version,
+                known_epoch,
+            },
+        );
+        self.send_request_and_decode(request).await
+    }
+
+    async fn get_new_transactions_with_proof(
+        &self,
+        known_version: Version,
+        known_epoch: Epoch,
+        include_events: bool,
+    ) -> Result<Response<(TransactionListWithProof, LedgerInfoWithSignatures)>> {
+        let request =
+            StorageServiceRequest::GetNewTransactionsWithProof(NewTransactionsWithProofRequest {
```

Comment:
Aah, because the size of the responses can be very different, e.g., outputs are already ~2x larger than transactions today (and most transactions aren't doing anything :P). This is to reduce the network data size for peers/clients that only want to execute the transactions.

Reply:
Yep, we have different syncing modes 😄 e.g., https://github.com/aptos-labs/aptos-core/blob/6899f9e34b99d0d2a9b9774142661dcabefb029f/config/src/config/state_sync_config.rs#L58